scripts/Muon/GUI/Common/grouping_tab_widget/grouping_tab_widget_presenter.py

```diff
@@ -154,10 +154,16 @@ def handle_load_grouping_from_file(self):
             if pair.forward_group in self._model.group_names and pair.backward_group in self._model.group_names:
                 self._model.add_pair(pair)
 
+        if default:
+            if default in self._model.group_names:
+                self._model.add_group_to_analysis(default)
+            elif default in self._model.pair_names:
+                self._model.add_pair_to_analysis(default)
+
```

Comment:
It shouldn't happen but if the file doesn't define defaults then there wont be any groups or pairs selected so maybe there should be an else block to catch it if it occurs at all.

Reply:
The case of a file not having a default it actually relatively common. In this case the call to plot_default_groups_or_pairs below sets the selected groups/pars to there default instead.

Reply:
Although this is not very clear from the code I will see if I can make it a bit less opaque.